src/main/java/DeleteCommand.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Class for the command that deletes a task from the tasklist.
+ */
+public class DeleteCommand extends Command {
+
+    private int taskNumber;
+
+    /**
+     * Constructor for the delete command.
+     *
+     * @param taskNum the index of the task to be deleted
+     */
+    public DeleteCommand(int taskNum) {
+        this.taskNumber = taskNum;
+    }
+
+    /**
+     * Indicates whether this is an exit command.
+     *
+     * @return false.
+     */
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Executes the delete command.
+     * Given task will be deleted from the tasklist at the index position.
+     * A completion message will be displayed after successful deletion.
+     *
+     * @param tasks tasklist that stores all the tasks
+     * @param ui ui object that deals with user interaction
+     * @param storage storage object that deals with writing into the text file
+     * @return string representation of the command response
+     * @throws DukeException exception is thrown when the index falls outside the number of tasks
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        try {
+            if (taskNumber > tasks.getNumOfTask() || taskNumber < 1) {
+                throw new DukeException("List only has " + tasks.getNumOfTask() + " items.");
```

Comment:
Should the DukeException be thrown within this try catch block within an if statement? Logic might be confusing. Maybe you can consider a way to catch this error and throw it below as well.